packages/gatsby-theme-store/src/sdk/search/useRegionalizedSearch.ts

```diff
@@ -0,0 +1,26 @@
+import { navigate } from 'gatsby'
+import { useLocation } from '@reach/router'
+import { useContext } from 'react'
+
+import { Context } from '../region/Provider'
+
+interface Options {
+  map: Maybe<string>
+}
+
+export const useRegionalizedSearch = (
+  { map }: Options,
+  staticPath: boolean
+) => {
+  const { pathname } = useLocation()
+  const region = useContext(Context)
+
+  if (region?.regionId === undefined || staticPath !== true) {
+    return
+  }
+
+  const newMap = map == null ? 'region' : `${map},region`
+  const to = `${pathname}/rId?map=${newMap}`
```

Comment:
what is this `rId` string? is this arbitrary or is it a magic string for some api?

Reply:
it's arbitrary. Just using it so we don't use the pre-rendered template/data and use the fallback `/__client_side_search__` instead